### DISCUSSION TITLE
Install and Setup

### DISCUSSION THREAD

**Discussion by bgowthaman** [eusonlito/GPS-Tracker#38] [Q&A]
I have too many errors on installation and setup, any one help me , or make installation video please

**Comment by eusonlito** [comment]
Do you follow requirements and installation instructions? And the errors are?

**Reply by bgowthaman** [reply]
yes, i am using centos 7. apache php mysql redis required version and extentions, file permission , SElinux permission, firewall port settings, some issus sovled, but i am stuck in error code 555 internal server error   "trim(): Argument #1 ($string) must be of type string, null given"   on monitor tab not working and device not connected

**Reply by eusonlito** [reply]
Send me full error log from `storage/logs/laravel/2024/02/15/2024-02-15.log`. Maybe some commando not available on CentOS.

**Reply by bgowthaman** [reply]
[2024-02-15.log](https://github.com/eusonlito/GPS-Tracker/files/14306164/2024-02-15.log)

**Reply by eusonlito** [reply]
I have updated the Monitor Panel, please update.

About the device, there are an error related with the port: `The server could not be started because the 8091 is not free`.

**Reply by bgowthaman** [reply]
[2024-02-16.log](https://github.com/eusonlito/GPS-Tracker/files/14314165/2024-02-16.log)
After update 'monitor > system' tab opened but "
It has not been possible to load the summary information" and 'monitor > installation' tab error "Undefined array key 0"
![image](https://github.com/eusonlito/GPS-Tracker/assets/86757140/594ee6df-dd20-4425-9e3e-db07495bfe25)
![image](https://github.com/eusonlito/GPS-Tracker/assets/86757140/bde84853-e8b7-4baa-88a3-aa02ba766f60)

**Reply by eusonlito** [reply]
I have updated the code, but I haven't tested the installation in a different OS than debian/ubuntu. I will test CentOS installation using docker.

**Reply by eusonlito** [reply]
I have updated the project to support centos 7 :)

Anyway you must install fuse command (from psmisc package) to allow to restart runing tracker servers.

**Reply by bgowthaman** [reply]
" Monitor Tab > System" ," Monitor Tab > Installation " working fine, But i unable get device connection, 502 bad gatway error on server incoming connection.
[2024-02-22.log](https://github.com/eusonlito/GPS-Tracker/files/14378422/2024-02-22.log)
[server-start-port-port-5055-reset-debug.log](https://github.com/eusonlito/GPS-Tracker/files/14378423/server-start-port-port-5055-reset-debug.log)

**Comment by eusonlito** [comment]
@bgowthaman one question, which protocol are you using?

**Reply by bgowthaman** [reply]
Osmand protocol @ 5055 port , I m testing traccar mobile client to connect GPS-Tracker

**Reply by eusonlito** [reply]
Update the code and test again, there was some problem with https server related with new features added on sockets.

**Reply by Bigbrian5** [reply]
@bgowthaman how did you get traccar mobile client working with GPS-Tracker?

**Reply by bgowthaman** [reply]
OsmAnd protocol and 5055 port server settings connect traccar mobile client , Actually my  GPS-Tracker refused incoming connection due to server configuration mistake ( CentOS7 Apache php MySQL Redis ), Best Method is Docker , i am testing Docker setup ... if GPS-Tracker public Docker image available all kinds of server configuration problem solved and very time save. If any one contribute public Docker repo for GPS- Tracker is helpful,

**Comment by Harwinder03** [comment]
Hi Lito im unable to install on aws ec2 ubuntu please share a guide or a vedio

**Reply by eusonlito** [reply]
Do you follow the installation guide? https://github.com/eusonlito/GPS-Tracker/wiki/%5BEN%5D-Local-Installation

Can you post your error?

**Reply by Harwinder03** [reply]
yes lito im on my system right now unable to go beyond composer error  
 -bash: ./composer: No such file or directory

**Reply by Harwinder03** [reply]
another error Your lock file does not contain a compatible set of packages. Please run composer update.

Script @composer install --no-scripts --no-dev --optimize-autoloader --classmap-authoritative --ansi handling the setup event returned with error code 2

**Reply by eusonlito** [reply]
Are you using docker?

**Reply by Harwinder03** [reply]
no its EC2 ubuntu 24

**Reply by eusonlito** [reply]
Are you sure you are inside the project directory?

**Reply by Harwinder03** [reply]
yes bro but composer gives diffrent errors

**Reply by eusonlito** [reply]
`composer` binary exists and is executable https://github.com/eusonlito/GPS-Tracker/blob/master/composer

**Reply by Harwinder03** [reply]
can we chat on telegram +919779274815

**Comment by Harwinder03** [comment]
can we chat on telegram +919779274815

**Reply by Harwinder03** [reply]
not able to get it working after many tries there is some issue with running the composer on php 8

**Reply by eusonlito** [reply]
Errors are "generic" and hidden or there are some error visible?

**Reply by Harwinder03** [reply]
Visible related to php library and sql conection also

**Reply by eusonlito** [reply]
Can you paste here the screenshot or error log?

**Reply by eusonlito** [reply]
Do you have checked the requirements? https://github.com/eusonlito/GPS-Tracker?tab=readme-ov-file#requirements

**Comment by Ricavasin** [comment]
Can anyone help me install and configure a local Ubuntu server? $$?
